pkg/controller/volume/persistentvolume/pv_controller.go

```diff
@@ -821,6 +826,12 @@ func (ctrl *PersistentVolumeController) updateClaimStatus(ctx context.Context, c
 				dirty = true
 			}
 		}
+
+		if utilfeature.DefaultFeatureGate.Enabled(features.VolumeAttributesClass) &&
+			!reflect.DeepEqual(claim.Status.CurrentVolumeAttributesClassName, volume.Spec.VolumeAttributesClassName) {
+			claimClone.Status.CurrentVolumeAttributesClassName = volume.Spec.VolumeAttributesClassName
```

Comment:
Let's imagine this scenario:

A pvc and pv have been bound with each other. And a vac (foo) is referenced by them. If a bad man updates the pv object with a different vac (bar). The pvc's currentVAC is still foo and never gets updated. With this change, once the status is changed, the csi-resizer will observe the mismatch and try to revert PV to the original.

Reply:
We can not send a warning event because the func is always called by `syncBoundClaim` in the every reconclie cycle after the claim is bound. If user changes the expected vac of the claim, they will recieve the warning message. the external-resizer can handle this change of the claim, so it will make users confused.

Another scenario is the static provisoning. If claim and volume has the same vac, no RPCs should be called by external-resizer, the expected behavior is to do binding operation. when the operation is completed, the claim's status is updated. we can see the vac in its status. During the process, external components should do nothing.

If the volume is dynamic provsioned, there will be a race between the pv controller and external components (provsioner, resizer) according to the current implementation.

Maybe we can only allow the pv controller to update the static provisoned volumes when pvc isn't matched with volume on the vac field.